### PR TITLE
docs(catalog): document the current set of well known annotations

### PR DESCRIPTION
**Pull request by freben**
#### :heavy_check_mark: Checklist

- [ ] All tests are passing `yarn test`
- [ ] Screenshots attached (for UI changes)
- [ ] Relevant documentation updated
- [ ] Prettier run on changed files
- [ ] Tests added for new functionality
- [ ] Regression tests added for bug fixes

### CODE REVIEW

File: docs/features/software-catalog/well-known-annotations.md [hunk 1]
```diff
@@ -0,0 +1,116 @@
...
+it even represents a single file. Note also that a single location can be the
+source of many entities, so it represents a many-to-one relationship.
+
+The format of the value is `<type>:<target>`. Note that the target may also
+contain colons, so it is not advisable to naively split the value on `:` and
+expecting a two-item array out of it. The format of the target part is
+type-dependent and could conceivably even be an empty string, but the separator
+colon is always present.
+
+### backstage.io/techdocs-ref
```
Comment (emmaindal) [inline docs/features/software-catalog/well-known-annotations.md:41]: 🎉 Thank you!!

File: docs/features/software-catalog/well-known-annotations.md [hunk 2]
```diff
@@ -0,0 +1,116 @@
+---
+id: well-known-annotations
+title: Well-known Annotations on Catalog Entities
+sidebar_label: Well-known Annotations
+---
+
+This section lists a number of well known
```
Comment (stefanalund) [inline docs/features/software-catalog/well-known-annotations.md:7]: This is a great start but I would like to see some more zoomed out and conceptually information. It should answer questions like:

* What are the purpose(s) of these annotations? 
* Who is defining them? 
* Why have well-known ones?
* How do I reason about namespacing?
* How are they consumed by plugins?
* Is this the only way to pass information between an entity and the plugin?

Maybe they even deserve it's own doc: `associating-plugins-to-exteranal-sources` or something similar.

Comment (freben) [reply docs/features/software-catalog/well-known-annotations.md:7]: Yes, I intended that to be a separate page. This one has only a very specific purpose, and it links out to just one other page (the descriptor format) for now, but more should be added.

**Review by stefanalund — APPROVED** [review]
Approving with the hope that the other page follows shortly 🤣

**Pull request merged**